### ISSUE TITLE
5k & 8k transcoding options

### ISSUE THREAD

**Issue by vid-bin** [Chocobozzz/PeerTube#5964]
### Describe the problem to be solved

Please add 5k and 8k transcoding options.

My setup is a bit unique because I use hardware encoding options and can transcode high resolution videos. I don't know how this is going to work for software h264. But peer tube should be pushing the envelope in terms of quality and if YouTube can do 8k transcoding then we should be able to do 8k transcoding.

**Comment by vid-bin** [comment]
Actually it should probably be 5k, 6k and 8k.

If you don't want to do this can you describe the steps needed to add it? I tried searching through the codebase for H_4K and 2160p and changed all of those files and peertube refuses to build, so there is probably another hidden setting in there.

**Comment by sawine** [comment]
Hello,

I would also really like the possibility of uploading up to 8k videos. We host 360 content and our footage can only look its best at 8K resolution.

Thanks

**Comment by dannekrose** [comment]
> ### Describe the problem to be solved
> Please add 5k and 8k transcoding options.
> 
> My setup is a bit unique because I use hardware encoding options and can transcode high resolution videos. I don't know how this is going to work for software h264. But peertube should be pushing the envelope in terms of quality and if YouTube can do 8k transcoding then we should be able to do 8k transcoding.
> 
> ### Describe the solution you would like
> _No response_

I agree more options are always better.  I think if the goal is to be "pushing the envelope," then 16k and 32k should be the goals.  16k video already exists. 32k displays are just around the corner, too.

https://vimeo.com/martinlisius

**Comment by dhk2** [comment]
This sounds good, but would increase the priority of a "max size" setting in redundancy configuration. 8k livestreams will quickly blow out an instance's local cache.

**Comment by dannekrose** [comment]
I personally don't have any plans to host content above 1080p60 for the near future due to storage costs, and I agree that being able to limit what the max size a peering instance will be willing to provide via redundancy would be really important before adding support for 4k and above.

**Comment by SimplyCorbett** [comment]
@Chocobozzz I'm just wondering if this is being considered. It would be nice to have at least 5k options available as there are cameras that record in 5k. Youtube used to have this option before they just went with 4k & 8k.

**Comment by AlexJade23** [comment]
Always higher and higher !
Is it really usefull ?
SONY's press service announced that 4K projectors are enough up to 12 meters screens for cinemas...
Who has a 12 meters screen at home ?

**Comment by vid-bin** [comment]
Random spam comment of the day goes to @AlexJade23!

I can answer that though! I have a 5k monitor! I'd like to be able to view my GoPro in its native 5k resolution rather than a scaled down 4k resolution.

**Comment by SimplyCorbett** [comment]
@Chocobozzz It's actually pretty easy to modify the source to add these options. Do you want me to submit a PR?

**Comment by Chocobozzz** [comment]
Sure but please add some tests too!

**Comment by sawine** [comment]
@SimplyCorbett  Could you please point me in the right direction to customize my Peertube instance to support 5k, 6k and 8k?

I've been searching but I can't find much information.  I saw the other thread about increasing the proxy limit to be able to playback 4k60 properly...

I'm surprised there isn't a plugin to add custom transcoding format?

Thanks

**Comment by SimplyCorbett** [comment]
@sawine I stopped using peertube (not because its bad, but because my use case changed). I may or may not start using it again in the future.

Back when I was using peertube all you had to do was use the transcoding-profile-debug plugin and then modify the source code of peertube to have higher resolution options. Use a code editor like notepad++ or sublime text and do a mass search on the entire peertube source folder for H_4K and 2160p. Then you need to add your custom resolutions line by line under (or on top) of each search result.

It’s actually not hard just a bit time consuming.

I don’t know how “hard” it would be to add such options to remote transcoding servers as all of my testing was done local with a GPU that does AV1 encoding.
